src/core/searcher.rs

```diff
@@ -89,6 +92,69 @@ impl Searcher {
         store_reader.get(doc_address.doc_id)
     }
 
+    /// Fetches documents from tantivy's store given a list of [`DocAddress`].
+    ///
+    /// This method is more efficient than calling [`doc`](Self::doc) multiple times, as it batches
+    /// overlapping requests to segments and blocks.
+    pub fn docs<D: DocumentDeserialize>(
```

Comment:
This function could be simplified, if we rely on the `BlockCache`. Since the doc_adresses are already sorted we can just call `doc` and always hit the cache.

Downside would be that this works only if the `BlockCache` parameter `cache_num_blocks` is not zero

Reply:
I don't love relying on this invariant, since the dependency is non-obvious, but I've left a comment. It certainly does simplify the code here, though.

In the future, we could also use a thread-pool to parallelize access to blocks, which may require us to revive the previous code but until then, this is fine.

Reply:
As written I don't think this pulls it weight as an API addition, especially since it enforces a data structure (`BTreeSet`) that stays sorted under modification instead of for example just sorting a slice of `DocAddress`es.

Maybe a performance hint on `Searcher::doc` to access multiple documents in address order if possible to improve cache hit rates would serve the same purpose?